src/evict/evict_lru.c

```diff
@@ -1374,29 +1385,31 @@ __evict_walk_file(WT_SESSION_IMPL *session,
 		/*
 		 * It's possible (but unlikely) to visit a page without a read
 		 * generation, if we race with the read instantiating the page.
-		 * Ignore those pages, but set the page's read generation here
-		 * to ensure a bug doesn't somehow leave a page without a read
-		 * generation.
+		 * Set the page's read generation here to ensure a bug doesn't
+		 * somehow leave a page without a read generation.
 		 */
-		if (page->read_gen == WT_READGEN_NOTSET) {
+		if (page->read_gen == WT_READGEN_NOTSET)
 			__wt_cache_read_gen_new(session, page);
```

Comment:
Another lurking policy change: as soon as we see a page without a read generation, we set it (to the middle of the range value), then immediately consider the page for eviction.  My thinking was that when scrubbing, we don't want to wait to walk the whole cache again for pages that aren't needed.